ignite/contrib/engine.py

```diff
@@ -0,0 +1,110 @@
+# coding: utf-8
```

Comment:
Perhaps even `trainers` and leave `engine` for ignite core.
  - engine -> class
  - trainer -> specific instance

Reply:
> Perhaps even trainers and leave engine for ignite core.

In `ignite.contrib.engines` we could later add evaluators too. Even in your case, do you need a supervised evalutor or native one is ok ?

Reply:
> we could later add evaluators too

That's a good point, even though we could have `ignite.contrib.evaluators` module.

In my case no, I'm doing without. There are cases in time series where you want to feed back your output into your input. This does not need to be defined outside of the model (i.e. could be done within the `forward` of `nn.Module` with a simple `if not self.training` because no optimization is done at this point) but if eventually someone write a code for that, it doesn't hurt to have it as an evaluator and not rewrite the code.